### PR TITLE
Use GitHub Actions to automatically label PRs

### PR DESCRIPTION
**Pull request by scholzj**
This PR tests how GitHub actions can be used to trigger automatic labeling of PRs based on their size and based on the files they changed.

### CODE REVIEW

File: .github/labeler.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  - cluster-operator/**/*
+
+topic operator:
+  - topic-operator/**/*
+
+user operator:
+  - user-operator/**/*
+
+docker images:
+  - docker-images/**/*
```
Comment (tombentley) [inline .github/labeler.yml]: I suppose we should have api in here too?

Comment (scholzj) [reply .github/labeler.yml]: I can add `api` as well if you want. But up until now we did not have `api` label at all so I'm not sure it is needed.

Comment (tombentley) [reply .github/labeler.yml]: I guess in the majority of cases something which touches api will touch on of the operators. The exception to that could be changes which touch `api` and only `operator-common`. Still I suspect this is good enough for now.

Comment (scholzj) [reply .github/labeler.yml]: Yeah, the way this works is that all matching labels are assigned. There will be almost no `api` only PRs. Most PRs changing the `api` will have it's part in one of the operators. We can always add more labels later.

**Pull request merged**